rules/java_rules.build_defs

```diff
@@ -84,6 +87,30 @@ def java_library(name:str, srcs:list=None, src_dir:str=None, resources:list=[],
 
         src_dir_label = [] if srcs else ['src_dir:'+src_dir]
 
+        def _jni_header_outputs(name, output):
+            for hdr_path in output:
+                add_out(name, hdr_path)
+                add_label(name, "cc:inc:" + join_path(package_name(), dirname(hdr_path)))
+
+        jni_hdrs = build_rule(
+            name = name,
+            tag = "jni_hdrs",
+            srcs = {
+                "WORKER": srcs or [src_dir],
+            },
+            cmd = " && ".join([
+                javac_cmd,
+                "find _jni -type f -name '*.h' | sort",
```

Comment:
This should be safe because `javac` only ever outputs on stderr by default, so it shouldn't mess up the post-build output collection.